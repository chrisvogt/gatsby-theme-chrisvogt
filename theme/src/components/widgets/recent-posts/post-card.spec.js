import React from 'react'
import renderer from 'react-test-renderer'
import PostCard from './post-card'

describe('PostCard', () => {
  const baseProps = {
    banner: 'https://cdn.example.com/images/article-og-banner.jpg',
    category: 'personal',
    date: '1592202624',
    link: '/blog/article',
    title: 'My Blog Post'
  }

  it('matches the snapshot without excerpt', () => {
    const tree = renderer.create(<PostCard {...baseProps} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders excerpt when provided', () => {
    const propsWithExcerpt = {
      ...baseProps,
      excerpt: 'This is a sample excerpt for the blog post.'
    }
    const tree = renderer.create(<PostCard {...propsWithExcerpt} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not render excerpt when explicitly null', () => {
    const propsWithNullExcerpt = {
      ...baseProps,
      excerpt: null
    }
    const tree = renderer.create(<PostCard {...propsWithNullExcerpt} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('does not render excerpt when explicitly undefined', () => {
    const propsWithUndefinedExcerpt = {
      ...baseProps,
      excerpt: undefined
    }
    const tree = renderer.create(<PostCard {...propsWithUndefinedExcerpt} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without banner when not provided', () => {
    const propsWithoutBanner = {
      category: 'personal',
      date: '1592202624',
      link: '/blog/article',
      title: 'My Blog Post'
    }
    const tree = renderer.create(<PostCard {...propsWithoutBanner} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders without category when not provided', () => {
    const propsWithoutCategory = {
      banner: 'https://cdn.example.com/images/article-og-banner.jpg',
      date: '1592202624',
      link: '/blog/article',
      title: 'My Blog Post'
    }
    const tree = renderer.create(<PostCard {...propsWithoutCategory} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('specifically tests excerpt prop handling for blog page change', () => {
    // This test specifically addresses the user's change where excerpt was commented out
    const propsWithoutExcerpt = {
      ...baseProps
      // excerpt is intentionally not included, simulating the commented out line
    }

    const component = renderer.create(<PostCard {...propsWithoutExcerpt} />)
    const tree = component.toJSON()

    // Verify that no excerpt paragraph is rendered
    const excerptParagraph = findExcerptParagraph(tree)
    expect(excerptParagraph).toBeNull()

    // Verify that the component still renders correctly without excerpt
    expect(tree).toBeTruthy()
  })

  it('verifies excerpt is rendered when explicitly provided', () => {
    const propsWithExcerpt = {
      ...baseProps,
      excerpt: 'This is a test excerpt that should be displayed.'
    }

    const component = renderer.create(<PostCard {...propsWithExcerpt} />)
    const tree = component.toJSON()

    // Verify that excerpt paragraph is rendered
    const excerptParagraph = findExcerptParagraph(tree)
    expect(excerptParagraph).toBeTruthy()
    expect(excerptParagraph.children).toContain('This is a test excerpt that should be displayed.')
  })
})

// Helper function to find the excerpt paragraph in the rendered tree
function findExcerptParagraph(tree) {
  if (!tree || !tree.children) return null

  for (const child of tree.children) {
    if (child.type === 'p' && child.props && child.props.className && child.props.className.includes('description')) {
      return child
    }
    if (child.children) {
      const found = findExcerptParagraph(child)
      if (found) return found
    }
  }
  return null
}
