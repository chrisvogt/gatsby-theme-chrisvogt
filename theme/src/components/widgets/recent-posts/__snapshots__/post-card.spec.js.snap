// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard matches the snapshot 1`] = `
<a
  className="css-sod6qt"
  href="/blog/article"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="css-139dwiw"
  >
    <div
      className="card-content css-x4n4mc"
    >
      <div
        className="card-media"
      >
        <div
          className="css-1hpd2dn"
        />
      </div>
      <div
        className="css-1blzwpa"
      >
        Personal
      </div>
      <h3
        className="css-13imqch"
      >
        My Blog Post
      </h3>
      <time
        className="created css-1c6hy0j"
      >
        1592202624
      </time>
    </div>
  </div>
</a>
`;
